documentation/images/plantuml/agent/state_machines/backhaul_manager_fsm.puml

```diff
@@ -0,0 +1,84 @@
+@startuml backahul_manager_fsm
+
+[*] --> INIT
+INIT --> WAIT_ENABLE
+WAIT_ENABLE --> ENABLED
+ENABLED --> MASTER_DISCOVERY : wired
+ENABLED --> WIRELESS_FSM : wireless
+state WIRELESS_FSM {
+    [*] --> INIT_HAL
+    INIT_HAL --> WPA_ATTACH
+    WPA_ATTACH --> CONNECT
+    CONNECT --> MASTER_DISCOVERY : connection success
+}
+MASTER_DISCOVERY --> OPERATIONAL
+OPERATIONAL --> RESTART : wired link status change
+RESTART --> INIT
+
+@enduml
+
+@startuml dynamic_backhaul_switch
+
+box "controller" #LightSlateGray
+participant controller
+end box
+
+box "prplmesh agent" #LightBlue
+participant backhaul_manager
+participant radio_agent
+end box
+
+box "platoform"
+participant hostapd
+participant supplicant
+end box
+
+== platform init ==
+hostapd <- : start disabled (static channel)
+activate hostapd
+deactivate hostapd
+supplicant <- : start (no network config)
+activate supplicant
+deactivate supplicant
+
+== init ==
+...
+== operational ==
+controller -> backhaul_manager : switch_backhaul(backhaul)
+activate backhaul_manager
+backhaul_manager -> backhaul_manager : restart
+note left
+m_preferred_backhaul = backhaul
+end note
+backhaul_manager -> supplicant : tear down
+note right
+disable all supplicants
+disable wired interface
+end note
+backhaul_manager -> radio_agent: disconnect
+deactivate backhaul_manager
+activate radio_agent
+radio_agent -> radio_agent : restart
+radio_agent -> hostapd : tear down
+deactivate radio_agent
+== init ==
+...
+backhaul_manager -> controller : autoconfig search
```

Comment:
Autoconfig search can only start after the backhaul has been established, so this has to go after the enable_interface.

Reply:
This is incorrect, we can start the autoconfig search without an active backhaul, and when the backhaul will be established, the CMDUs will arrive at the controller.
This is IMO a better design since it doesn't require ordering between the WPS and the onboarding operations - in fact, the UCC first sends dev_set_config, and only after does it trigger WPS - so your suggestion will not work in certification.

Reply:
Let me rephrase: autoconfig search can only reach the controller once backhaul is established :-)

So to be fully correct, it would have to be:
```
backhaul_manager -> backhaul_manager : go to autoconfig search state
... (set up backhaul)
backhaul_manager -> controller : autoconfig search
```
But OK this complicates things, and it's better explained in text anyway.

Reply:
eventually, it is as you said in the first place (probably since you implemented it :) ):
```
backhaul_manager -> backhaul_manager : set up backhaul
ucc -> backhaul_manager : start_wps_pbc
... (establish backhaul connection vi wps_pbc)
backhaul_manager -> controller : autoconfig search
```